presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -17,9 +14,14 @@
 
 class AnalyzerEngine(analyze_pb2_grpc.AnalyzeServiceServicer):
 
-    def __init__(self, registry=RecognizerRegistry(),
-                 nlp_engine=SpacyNlpEngine()):
+    def __init__(self, registry,
```

Comment:
Not sure what the right answer is. In my opinion we should set up the default objects if nothing is passed.
```
def __init__(self, registry=None, nlp_engine=None):
    if not registry:
        registry = RecognizerRegistry()
    if not nlp_engine:
        nlp_engine = SpacyNlpEngine()
```

This might make life easier for someone who's trying to initialize this class but isn't sure what the required input is.

Reply:
that point was raised during code review with itye, and i agree with him that the right DI pattern is not to instantiate the injected component within the type. this way analyzer_engine does not know of recognizerregistry and spacynlpengine concrete types, which is the right approach.
perhaps a static helper/factory method is what we want?

Reply:
WDYT about this option?

```
   def __init__(self, registry=None, nlp_engine=None):
        if not registry:        
            from analyzer import RecognizerRegistry 
            registry = RecognizerRegistry()
        if not nlp_engine:
            from analyzer.nlp_engine import SpacyNlpEngine 
            nlp_engine = SpacyNlpEngine()
```